pkg/alertmanager/alertmanager_template.go

```diff
@@ -95,7 +96,7 @@ func WithCustomFunctions(userID string) template.Option {
 
 // loadTemplates produces a template.Template from several in-memory template files.
 // It is adapted from FromGlobs in prometheus/alertmanager: https://github.com/prometheus/alertmanager/blob/9de8ef36755298a68b6ab20244d4369d38bdea99/template/template.go#L67-L95
-func loadTemplates(tmpls []io.Reader, options ...template.Option) (*template.Template, error) {
+func loadTemplates(tmpls []alertingTemplates.TemplateDefinition, options ...template.Option) (*template.Template, error) {
```

Comment:
[nit] I don't think the `grafana/alerting` types need to leak in here, I think this is cleaner.

```suggestion
func loadTemplates(tmpls []string, options ...template.Option) (*template.Template, error) {
```